Compile error with Visual Studio 2017 when using c++17 or later

Hello,

If I build master with cmake Visual Studio 15 2017 generator arch x64 on windows 10, the examples build fine.
If I change the C++ language standard to c++17 (or later) in the Visual Studio UI (Configuration Properties -> C/C++ -> Language -> C++ Language Standard) and then try to rebuild, I get the following error:

```
1>------ Build started: Project: read_write_dataset_string_bin, Configuration: Debug x64 ------
1>read_write_dataset_string.cpp
1>c:\dev\highfive\include\highfive\bits/H5Path_traits_misc.hpp(31): error C2248: 'HighFive::Object::Object': cannot access protected member declared in class 'HighFive::Object'
1>c:\dev\highfive\include\highfive\bits/H5Object_misc.hpp(19): note: see declaration of 'HighFive::Object::Object'
1>c:\dev\highfive\include\highfive\H5Object.hpp(36): note: see declaration of 'HighFive::Object'
1>c:\dev\highfive\include\highfive\bits/H5Path_traits_misc.hpp(19): note: while compiling class template member function 'HighFive::PathTraits<HighFive::DataSet>::PathTraits(void)'
1>c:\dev\highfive\include\highfive\bits\../H5DataSet.hpp(99): note: see reference to function template instantiation 'HighFive::PathTraits<HighFive::DataSet>::PathTraits(void)' being compiled
1>c:\dev\highfive\include\highfive\bits\../H5DataSet.hpp(31): note: see reference to class template instantiation 'HighFive::PathTraits<HighFive::DataSet>' being compiled
1>Done building project "read_write_dataset_string_bin.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

It seems that adding `PathTraits` as a friend class of `Object` fixes this issue (see linked PR).

--- a/include/highfive/H5Object.hpp
+++ b/include/highfive/H5Object.hpp
@@ -93,6 +93,8 @@ class Object {
     friend class AnnotateTraits;
     friend class Reference;
     friend class CompoundType;
+    template <typename Derivate>
+    friend class PathTraits;
 };
 
 

